Dockerfile

```diff
@@ -60,17 +61,19 @@ RUN go install github.com/bazelbuild/buildtools/buildozer@${BDR_VER} \
   && rm -rf $(go env GOCACHE) \
   && rm -rf $(go env GOMODCACHE)
 
-WORKDIR ${GOPATH}/src/github.com/envoyproxy/protoc-gen-validate
-COPY . .
-
 # python must be on PATH for the execution of py_binary bazel targets, but
 # the distribution we installed doesn't provide this alias
 RUN ln -s /usr/bin/python3.8 /usr/bin/python
 
+WORKDIR ${GOPATH}/src/github.com/envoyproxy/protoc-gen-validate
+
 # python tooling for linting and uploading to PyPI
+COPY requirements.txt .
```

Comment:
Optimized the docker build to benefit from layer caching so we don't re-download the Python dependencies if any other files change.